### PR TITLE
V4: remove `sockjs_client` and some deprecated APIs

### PR DESCRIPTION
**Pull request by evanweible-wf**
## Motivation
- In order to publish any further releases, we have to remove our dep on `sockjs_client` since it is a git reference (this is enforced by https://pub.dev).

## Changes
- Remove `sockjs_client` dep and all usages
- Remove deprecated APIs that we know to be unused internally
- Update changelog

### CODE REVIEW

**Comment by aviary2-wf** [comment]
## Security Insights
The items listed below may not capture all security relevant changes. Before providing a security review, be sure to review the entire PR for security impact.
<details>
<summary>(2) Security relevant changes were detected</summary>
<li>Watched keyword <code>withCredentials</code> in <code>lib/src/http/client.dart</code> line(s) ['24', '64'] removed</li>
<li>Watched keyword <code>withCredentials</code> in <code>lib/src/http/http_client.dart</code> line(s) ['81'] added</li>
</details>

## Action Items
* Obtain a [security review](https://wiki.atl.workiva.net/display/SECURITY/Development+Security+Review+Guidelines); reviewer should pay special attention to insights listed above
* Verify [`aviary.yaml`](https://dev.workiva.net/docs/teams/information-security/aviary#configuring-aviary) coverage of security relevant code
---
_Questions or Comments? Reach out on Slack: #support-infosec._

File: lib/src/mocks/mock_web_socket_server.dart
Comment (corwinsheahan-wf) [inline lib/src/mocks/mock_web_socket_server.dart]: Nice!

File: pubspec.yaml
Comment (corwinsheahan-wf) [inline pubspec.yaml]: woohoo!

**Comment by maxwellpeterson-wf** [comment]
+1 security

File: CHANGELOG.md
Comment (greglittlefield-wf) [inline CHANGELOG.md:39]: It looks like the `polly-backoff` reference link doesn't exist

File: lib/src/web_socket/browser/sockjs.dart
Comment (greglittlefield-wf) [inline lib/src/web_socket/browser/sockjs.dart:33]: #nit this comment is a little out of date

>The SockJS wrapper library is preferred [over the SockJS Dart port]

But the port is no longer an option, so we might want to reword this

**Review by greglittlefield-wf — APPROVED** [review]
A nit and a broken link in the changelog, but nothing worth holding up the merge of this PR.

+1

**Review by greglittlefield-wf — DISMISSED** [review]

**Comment by evanweible-wf** [comment]
CI will fail on stable/dev/beta because of some outdated dependencies. We'll ignore that for now and cut a separate release with more dependency updates after we cut v4.

**Comment by evanweible-wf** [comment]
QA +1
- [x] CI passing
- [x] Consumer tested this branch against all w_transport consumers within the Workiva org

**Review by greglittlefield-wf — APPROVED** [review]

**Comment by evanweible-wf** [comment]
@Workiva/release-management-p

**Review by rmconsole-wf — APPROVED** [review]
+1 from RM

**Pull request merged**
